okta_rules/okta_geo_improbable_access.py

```diff
@@ -52,8 +52,8 @@ def rule(event):
     # Calculation is complete, so store the most recent login for the next check
     store_login_info(login_key, event)
     EVENT_CITY_TRACKING[event['p_row_id']] = {
-        'new_city': new_login_stats['city'],
-        'old_city': old_login_stats['city'],
+        'new_city': new_login_stats.get('city', 'None'),
+        'old_city': old_login_stats.get('city', 'None'),
```

Comment:
```suggestion
        'new_city': new_login_stats.get('city', 'Not Found'),
        'old_city': old_login_stats.get('city', 'Not Found'),
```

Maybe `Not Found` would be more clear? They're (probably) not logging from no city, we just don't have info on what city they're logging in from.